nginx.tmpl

```diff
@@ -121,35 +178,20 @@ server {
 {{ $is_regexp := hasPrefix "~" $host }}
 {{ $upstream_name := when $is_regexp (sha1 $host) $host }}
 
-# {{ $host }}
-upstream {{ $upstream_name }} {
-
-{{ range $container := $containers }}
-	{{ $addrLen := len $container.Addresses }}
-
-	{{ range $knownNetwork := $CurrentContainer.Networks }}
-		{{ range $containerNetwork := $container.Networks }}
-			{{ if (and (ne $containerNetwork.Name "ingress") (or (eq $knownNetwork.Name $containerNetwork.Name) (eq $knownNetwork.Name "host"))) }}
-				## Can be connected with "{{ $containerNetwork.Name }}" network
-
-				{{/* If only 1 port exposed, use that */}}
-				{{ if eq $addrLen 1 }}
-					{{ $address := index $container.Addresses 0 }}
-					{{ template "upstream" (dict "Container" $container "Address" $address "Network" $containerNetwork) }}
-				{{/* If more than one port exposed, use the one matching VIRTUAL_PORT env var, falling back to standard web port 80 */}}
-				{{ else }}
-					{{ $port := coalesce $container.Env.VIRTUAL_PORT "80" }}
-					{{ $address := where $container.Addresses "Port" $port | first }}
-					{{ template "upstream" (dict "Container" $container "Address" $address "Network" $containerNetwork) }}
-				{{ end }}
-			{{ else }}
-				# Cannot connect to network of this container
-				server 127.0.0.1 down;
-			{{ end }}
-		{{ end }}
+{{ $paths := groupBy $containers "Env.VIRTUAL_PATH" }}
```

Comment:
May I ask you to change this line like in this https://github.com/abak-press/nginx-proxy/blob/760b3321a4e25662189572ed5a41fc1574138397/nginx.tmpl#L177
It's only one line, but gives us a huge profit.

We have a working image with this change https://hub.docker.com/r/abakpress/nginx-proxy/

Reply:
@bibendi @RafaelKr 
Can you explain please a bit more what the comma does (you can even explain the groupByMulti)?
Remember that I am not experienced in this tmpl go language and this only a "inherited" PR for me.

I have been reading https://github.com/jwilder/docker-gen/blob/master/template.go
And I can guess that with the change you are proposing a host can have a comma separated list of VIRTUAL_PATHs and the template will generate a upstream definition for each one( and also a location).

Am I right?

Thank you for your contribution

Reply:
You are right. With this change https://github.com/abak-press/nginx-proxy/commit/fb9c2f1328bfe3ebbdd587b94f8ec91934bcb3f5#diff-6c19a779564e1e2f88854a4a2fffbb78R170 we able to pick multiple paths for one host.

Real world docker-compose example:

```yaml
networks:
  - default
  - frontend
dns: $DIP_DNS
environment:
  - VIRTUAL_HOST=*.$COSMOS_PROJECT.docker
  - VIRTUAL_PATH=/cosmos,/cosmos-assets,/dashboard/seller,/dashboard/customer,/api/v1/seller,/api/v1/customer
````

Reply:
Without that change it still works using  VIRTUAL_PATH="~^/(path1|path2)"

Reply:
@w4rri0r3k That solution is way better in my opinion. I will add it to the readme